### ISSUE TITLE
Cache always used even with branches

### ISSUE THREAD

**Issue by kingcrunch** [composer/composer#1571]
The problem I faced is, that when I have a dependency to a dev-version (--> a branch) and I update the dependencies on my local machine, the new revisions weren't installed at all.
On the remote machine the `vendor`-folder exists and the (old) package is within it. When I call `install` it does _not_ update this package even if the packages vcs-ref differs from the installed one. When I remove the `vendor`-folder and call `install` again everything works fine. When I remove the `vendor`-folder and call `install` again it successfully fetches everything from the cache.

TL;DR: Caching doesn't work with branch-based-dependencies, when the package is already installed in a different revision.

As a sidenote: Just realize, that it is possible, that this behaviour is not related to caching.
